docs/schulcloud-server/Coding-Guidelines/modules-submodules.md

```diff
@@ -0,0 +1,66 @@
+# Implementation and usage of modules, submodule and barrel files in our project
+
+In this guide, we'll cover how to use modules, submodules and barrel files in our project. These concepts help you organize your code into separate files and directories, making it easier to manage and maintain.
+
+## Modules and Submodules
+
+In our project, modules are a way to create separate scopes. This means that interfaces, use cases, services, etc., declared in a module are not visible outside the module unless they are explicitly exported using the `export` keyword. To import a module, you use the `import` keyword followed by the module name. Here's an example:
+
+```typescript
+import { ModuleName } from '@modules/module-name';
+```
+
+Submodules are modules that are part of a larger module. They should be used within the main module. If it is necessary to use parts of the submodule outside the main module, the main module should export this via its barrel file(index.ts):
+
+
+```typescript
+// @modules/module-name/index.ts
+export { SubmoduleServiceName } from './submodule-name/service.ts';
+```
+
+## Barrel Files
+
+Barrel files are a way to rollup exports from several modules into a single convenient module. The barrel itself is a module file that re-exports selected exports of other modules.
+
+If you have several related modules in a directory, you can create a barrel to re-export all of their exports. This allows other modules to import everything from the barrel instead of having to import things individually from each module.
+
+Here's an example of a barrel file:
+
+```typescript
+// @modules/module-name/index.ts
+export { PublicService } from './services/public-service.ts';
+export * from './interfaces';
```

Comment:
Wildcards wollen wir nicht in den module barrel files. Ich glaube da müssen wir auch explzit differenzierung und begründen warum es genau dort nicht gewünscht ist.